client/state/themes/reducer.js

```diff
@@ -474,18 +474,19 @@ export const themeFilters = withSchemaValidation( themeFiltersSchema, ( state =
  * Returns updated state for recommended themes after
  * corresponding actions have been dispatched.
  *
- * @param  {object} state  Current state
- * @param  {object} action Action payload
+ * @param   {object} state  Current state
+ * @param   {object} action Action payload
  * @returns {object}        Updated state
  */
-export function recommendedThemes( state = { isLoading: true, themes: [] }, action ) {
+export function recommendedThemes( state = {}, action ) {
+	const key = () => camelCase( action.filter );
```

Comment:
Is there any particular reasons to have this as a function rather than just a simple variable?
E.g.
```js
const key = camelCase( action.filter );

return { ...state, [ key ]: { /* ... */ } };
```

Reply:
I had it that way initially but then I realized that, since this was a reducer, I would be `camelCase`ing `action.filter` on every single state change and I didn't want to do that for perf reasons.

Reply:
This seems like premature perf optimization. I did a `console.time()` and `console.endTime()` on L481.5 and L491 and didn't see any appreciable differences. Maybe I'm not understanding the use case where we'd see the benefits?

Reply:
What if we didn't `camelCase` at all?

We are still using the array notation anyway (e.g. `[ key() ]`, `recommendedThemes[ key ]`), so we might as well embrace the hyphenated names and call it a day. 😅 

(It's not very common in the Calypso state, but I have found several occurrences of hyphenated properties here and there. See `productsList.items` for example.)